### PR TITLE
fix(notification) delete dependencies 

### PR DESCRIPTION
**Pull request by loiclau**
## Description

we need to delete any depencies that are orphaned when deleting the last “parent” service.

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [x] 19.04.x
- [x] 19.10.x
- [x] 20.04.x
- [x] 20.10.x (master)

<h2> How this pull request can be tested ? </h2>

    Add 2 services,
    Add a service dependency from 'Configuration > Notifications > Services' with one service as parent and the other as child,
    Delete the service you just put as parent,
    Go back to dependency menu,
    The dependency does not appear anymore.

## Checklist

- [x] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/class/centreon-clapi/centreonHostGroup.class.php [hunk 1]
```diff
@@ -123,6 +124,7 @@ public function initInsertParameters($parameters = null)
         $this->checkParameters();
     }
 
+
```
Comment (jeremyjaouen) [inline www/class/centreon-clapi/centreonHostGroup.class.php:127]: remove empty line

Comment (loiclau) [inline www/class/centreon-clapi/centreonHostGroup.class.php:127]:
```suggestion
```

**Review by jeremyjaouen — DISMISSED** [review]

File: lib/Centreon/Object/Dependency/DependencyServiceParent.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ * For more information : contact@centreon.com
+ *
+ */
+
+require_once "Centreon/Object/Object.php";
+
+/**
+ * Used for interacting with dependencies
+ *
+ * @author sylvestre
```
Comment (sc979) [inline lib/Centreon/Object/Dependency/DependencyServiceParent.php:27]: who is sylvestre ?

Comment (sc979) [inline lib/Centreon/Object/Dependency/DependencyServiceParent.php:27]:
```suggestion
```

File: lib/Centreon/Object/Dependency/DependencyServicegroupParent.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+ * For more information : contact@centreon.com
+ *
+ */
+
+require_once "Centreon/Object/Object.php";
+
+/**
+ * Used for interacting with dependencies
+ *
+ * @author sylvestre
```
Comment (sc979) [inline lib/Centreon/Object/Dependency/DependencyServicegroupParent.php:27]:
```suggestion
```

File: www/class/centreon-clapi/centreonServiceGroup.class.php [hunk 4]
```diff
@@ -395,6 +396,22 @@ public function __call($name, $arg)
         }
     }
 
+    /**
+     * Del Action
```
Comment (sc979) [inline www/class/centreon-clapi/centreonServiceGroup.class.php:400]:
```suggestion
     * Delete Action
```

File: www/include/configuration/configObject/hostgroup/DB-Func.php [hunk 5]
```diff
@@ -1,4 +1,5 @@
 <?php
+
 /*
  * Copyright 2005-2019 Centreon
```
Comment (sc979) [inline www/include/configuration/configObject/hostgroup/DB-Func.php:4]:
```suggestion
 * Copyright 2005-2020 Centreon
```

**Review by sc979 — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Pull request merged**